common-lib/src/main/scala/com/gu/mediaservice/GridClient.scala

```diff
@@ -132,73 +94,91 @@ class GridClient(apiKey: String, services: Services, maxIdleConnections: Int, de
     }
   }
 
-  private def makeRequestAsync(url: URL, apiKey: String): Future[Response] = {
-    val request = new Request.Builder().url(url).header(apiKeyHeaderName, apiKey).build
-    val promise = Promise[Response]()
-    httpClient.newCall(request).enqueue(new Callback {
-      override def onFailure(call: Call, e: IOException): Unit = promise.failure(e)
+}
+
+class GridClient(services: Services)(implicit wsClient: WSClient) extends LazyLogging {
 
-      override def onResponse(call: Call, response: Response): Unit = promise.success(response)
-    })
-    promise.future
+  def makeGetRequestAsync(url: URL, authFn: WSRequest => WSRequest)
+                         (implicit ec: ExecutionContext): Future[Response] = {
+    val request: WSRequest = wsClient.url(url.toString)
+    val authorisedRequest = authFn(request)
+    authorisedRequest.get().map { response => validateResponse(response, url)}
   }
 
-  private def validateResponse(res: ResponseWrapper, url: URL): Unit = {
-    import res._
-    if (statusCode != 200 && statusCode != 404) {
-      val errorMessage = s"breaking the circuit of full image projection, downstream API: $url is in a bad state, code: $statusCode"
-      val downstreamErrorMessage = res.bodyAsString
+  private def validateResponse(
+                                   response: WSResponse,
+                                   url: URL
+                                 ): Response = {
+//    val res = processResponse(response, url)
```

Comment:
Nice to see this method disappear. Let's also get rid of the comment.

Reply:
Done.